src/byteorder.rs

```diff
@@ -498,37 +594,74 @@ define_type!(
     u64,
     64,
     8,
-    read_u64,
-    write_u64,
+    u64::from_be_bytes,
+    u64::to_be_bytes,
+    u64::from_le_bytes,
+    u64::to_le_bytes,
     "unsigned integer",
     [u128],
     [u128],
     [U128],
     [U128]
 );
-define_type!(A, U128, u128, 128, 16, read_u128, write_u128, "unsigned integer", [], [], [], []);
+define_type!(
+    A,
+    U128,
+    u128,
+    128,
+    16,
+    u128::from_be_bytes,
+    u128::to_be_bytes,
+    u128::from_le_bytes,
+    u128::to_le_bytes,
+    "unsigned integer",
+    [],
+    [],
+    [],
+    []
+);
+define_type!(
+    A,
+    Usize,
+    usize,
+    mem::size_of::<usize>() * 8,
```

Comment:
The rustdoc CI failures seem to come from the fact that, in `define_type!`, we use `stringify!` on this `expr` in order to generate the doc comment.

I think the simplest thing to do here might just be to introduce a new macro argument that is used in the summary sentence of the doc comment instead of doing `stringify!($bits)`. It would have the text, "A 16-bit unsigned integer" , "A word-sized signed integer", etc. i.e.:

### Old

```rust
        doc_comment! {
            concat!("A ", stringify!($bits), "-bit ", $number_kind,
            " stored in a given byte order.
```

### New

```rust
        doc_comment! {
            concat!($description, " stored in a given byte order.
```

This would also allow us to get rid of the `$number_kind` argument.